dotnet/src/SemanticKernel.Abstractions/Contents/ImageContent.cs

```diff
@@ -16,6 +17,11 @@ public sealed class ImageContent : KernelContent
     /// </summary>
     public Uri? Uri { get; set; }
 
+    /// <summary>
+    /// The Data used as DataUri for the image.
+    /// </summary>
+    public BinaryData? Data { get; }
+
     /// <summary>
```

Comment:
But which has higher priority uri or base64 and why?

Reply:
As the `ToString` override suggests, Data has priority. This is also assured by Tests

Reply:
`ToString()` rather will not used in connectors. We should clearly explain to others that they should prioritize a BinaryData over an Uri. For example by adding additional information to the doc comment in ImageContent class.

Reply:
I was also looking at an issue opened in runtime asking to remove the length limitation on Uri. I think this is in general a good idea, but this won't be GA before net9 and this is way too late. SK and AI development is moving so quickly right now, that most wouldn't want to wait for that change.

But even with that change, I think `BinaryData` gives developers a lot of comfortable possibilities and they don't have to do all the base64 encoding and serialization themselfs. So maybe once the limit on Uri is removed, we could in the implementation just switch back to `Uri` as an implementation detail. I really hope the issue on `Azure.AI.OpenAI` is moving forward sooner rather than later, so we can finally merge this change.